org.civicrm.bootstrapcivicrm/CRM/Contribute/Form/ContributionCharts.php

```diff
@@ -198,6 +198,28 @@ public function postProcess() {
       }
       $values['size'] = array('xSize' => $xSize, 'ySize' => $ySize);
     }
+
```

Comment:
can you please extract the whole logic you have done here into a private class method ( e.g : styleCharts() or whatever you like to call it ) ..

```php
 $chartData = styleCharts($chartData);
```

```
private function styleCharts($chartData)  {
//stuff here
}
```